spec/models/settings/base_spec.rb

```diff
@@ -0,0 +1,169 @@
+require "rails_helper"
+
+RSpec.describe Settings::Base, type: :model do
+  with_model :TestSetting, superclass: described_class do
```

Comment:
Here we are specifying that we want a new model called `TestSetting` which descends from `Settings::Base`.